src/screens/Feed.js

```diff
@@ -1,124 +1,122 @@
-/* eslint-disable react-native/no-inline-styles */
 import React from 'react';
-import {View, StyleSheet, FlatList} from 'react-native';
+import {
+  View,
+  StyleSheet,
+  FlatList,
+  TouchableWithoutFeedback,
+  Text,
+  SafeAreaView,
+  Alert,
+} from 'react-native';
+
 import SettingsButton from './app/Components/SettingsButton/SettingsButton';
 import ProfileAvatar from './app/Components/ProfileAvatar/ProfileAvatar';
 import Post from './app/Components/Post/Post';
+import Swiper from 'react-native-swiper';
+import {Divider} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
+import * as ScreenshotDetector from 'react-native-screenshot-detect';
+
 export default class FeedScreen extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      swiperIndex: 0,
+      posts: [],
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchPosts();
+
+    this.eventEmitter = ScreenshotDetector.subscribe(() => {
+      Alert.alert(
+        'The post with caption',
+        posts[this.state.swiperIndex].caption,
+        'was screenshotted',
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    ScreenshotDetector.unsubscribe(this.eventEmitter);
   }
 
+  fetchPosts = () => {
+    const subscriber = firestore()
+      .collection('posts')
+      .onSnapshot(querySnapshot => {
```

Comment:
Do we have any .error() promises, for catching errors? Ie what if the firestore doesn't subscribe

Reply:
.catch(function(error) {
    console.error("Error connecting to firebase: ", error);
});

.catch(function(error) {
    console.error("Error finding posts collection: ", error);
});
.catch(function(error) {
    console.error("Error pulling in posts: ", error);
});